### PR TITLE
Enable OperatingSystemManager by default

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What does this PR do / Why do we need it**:
Operating System Manager is enabled by default and it's responsible for creating and managing the required configurations for worker nodes. Users can disable OSM per cluster basis.

Previously, since it was an experimental feature we had the `OperatingSystemManager` feature gate for it. But since it's considered GA now, we are going to remove that feature gate. 

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
* Operating System Manager is enabled by default and it's responsible for creating and managing the required configurations for worker nodes. 
* [ACTION REQUIRED] For existing clusters, users need to set `enableOperatingSystemManager` to true in the cluster spec to enable OSM. Existing `MachineDeployments` will not be rotated automatically. To use OSM for existing `MachineDeployments`, the user needs to update the `MachineDeployments` manually. An example for a somewhat benign change that could trigger rotation is to update the .spec.templates.metadata.annotations field of a MachineDeployment. This would result in the annotation being added to the machines and the machines would be rotated.
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/hold

I'll remove the hold myself.

**Review by xrstf — APPROVED** [review]
/approve

File: pkg/install/stack/kubermatic-seed/migrations.go [hunk 1]
```diff
@@ -118,6 +120,28 @@ func deleteLogrotateAddon(ctx context.Context, kubeClient ctrlruntimeclient.Clie
...
+			err := kubeClient.Update(ctx, &cluster)
+			if err != nil {
+				errors = append(errors, err)
+			}
+		}
+	}
+
+	return kerrors.NewAggregate(errors)
+}
+
```
Comment (ahmedwaleedmalik) [inline pkg/install/stack/kubermatic-seed/migrations.go]: The migration using installer.

File: pkg/resources/machinecontroller/webhook.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func WebhookDeploymentCreator(data machinecontrollerData) reconciling.NamedDeplo
 			}
 
 			// Enable validations corresponding to OSM
-			if data.Cluster().Spec.EnableOperatingSystemManager {
+			if data.Cluster().Spec.EnableOperatingSystemManager == nil || *data.Cluster().Spec.EnableOperatingSystemManager {
```
Comment (xrstf) [inline pkg/resources/machinecontroller/webhook.go]: Couldn't this also use `data.Cluster().Spec.IsOperatingSystemManagerEnabled()` ?

**Review by xrstf — APPROVED** [review]
/approve

Yay, my last OSM PR!

File: pkg/resources/machinecontroller/webhook.go (hunk 2 above)
Comment (ahmedwaleedmalik) [reply pkg/resources/machinecontroller/webhook.go]: Good point, refactored.

**Comment by xrstf** [comment]
/override pre-kubermatic-mla-e2e

fixed via #10565

**Comment by xrstf** [comment]
/hold

temporarily

**Comment by xrstf** [comment]
/hold

temporarily. also i see other PRs that have no problem with creating hetzner clusters, so either their API is still flaking or this PR actually broke hetzner support somehow.

**Comment by ahmedwaleedmalik** [comment]
> aec163a

Umm that's highly unlikely since everything worked fine till https://github.com/kubermatic/kubermatic/pull/10415/commits/aec163a3039b2f14ee87bd639d57438ff461e9f8

But we never know 🤷‍♂️

**Comment by ahmedwaleedmalik** [comment]
```
    hetzner_test.go:137: failed to create cluster: [POST /api/v1/projects/{project_id}/dc/{dc}/clusters][403] createClusterForbidden 
--- FAIL: TestCreateUpdateHetznerCluster (20.83s)
    --- FAIL: TestCreateUpdateHetznerCluster/create_cluster_on_Hetzner (20.83s)
```

🤔

**Comment by xrstf** [comment]
The same Prowjob also, if you just scroll up, did

```
--- PASS: TestCreateUpdateHetznerCluster (578.17s)
    --- PASS: TestCreateUpdateHetznerCluster/create_cluster_on_Hetzner (578.17s)
```

So I doubt this is purely Hetzner related.

**Comment by xrstf** [comment]
Seems like this time the tests are passing.

**Pull request merged**